### ISSUE TITLE
postinstall": "pnpm dlx sherif@latest" throws an error when installing dev dependecies

### ISSUE THREAD

**Issue by davidfers** [QuiiBz/sherif#45]
Having  ` "postinstall": "pnpm dlx sherif@latest",`   for linting when installing a new package, it throws an error only when the new package is installed as dev dependency. If I do pnpm dlx sherif@latest manually after package is installed it's all ok

the log
```
 ERR_INVALID_ARG_TYPE  The "path" argument must be of type string. Received undefined
│ pnpm [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
│     at new NodeError (node:internal/errors:405:5)
│     at validateString (node:internal/validators:162:11)
│     at Object.join (node:path:1171:7)
│     at getBinName (/usr/lib/node_modules/pnpm/dist/pnpm.cjs:210074:37)
│     at Object.handler [as dlx] (/usr/lib/node_modules/pnpm/dist/pnpm.cjs:210050:54)
│     at async /usr/lib/node_modules/pnpm/dist/pnpm.cjs:219832:21
│     at async main (/usr/lib/node_modules/pnpm/dist/pnpm.cjs:219799:34)
│     at async runPnpm (/usr/lib/node_modules/pnpm/dist/pnpm.cjs:220056:5)
│     at async /usr/lib/node_modules/pnpm/dist/pnpm.cjs:220048:7
│  ELIFECYCLE  Command failed with exit code 1.
```

**Comment by QuiiBz** [comment]
Looking at the stacktrace, the error seems to be coming from pnpm itself?

I've tried on a basic monorpeo and it works fine, could you share more information about your Node & pnpm version, and provide a minimal reproduction?

**Comment by davidfers** [comment]
node: v18.18.2
pnpm: 8.12.0

I have tested it with turborepo using pnpm
`npx create-turbo@latest`

create script 
`"postinstall" : "pnpm dlx sherif@latest"` 

`pnpm -F docs add -D tailwindcss-animate`

```
...
> pnpm dlx sherif@latest
│ Progress: resolved 1, reused 0, downloaded 0, added 0
│ Packages: +2
│ ++
│ Progress: resolved 7, reused 2, downloaded 0, added 2, done
│  ERR_INVALID_ARG_TYPE  The "path" argument must be of type string. Received undefined
│ pnpm [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
```
I noticed that using `"postinstall" : "npx sherif@latest"`  is working fine

**Comment by QuiiBz** [comment]
Forgot to ask what's your OS? I'm using macOS and still can't reproduce using the same versions. Could you share a minimal reproduction (I believe it can be a single `package.json`)?

Do you reproduce when using something else than `sherif`, e.g. `pnpm dlx husky-init`? The issue doesn't seems to be coming from Sherif itself.

**Comment by davidfers** [comment]
I'm using linux but I found [here](https://github.com/t3-oss/create-t3-turbo/issues/763) that is happening on mac too.
As stated in the issue I pasted above, it can be reproduced starting a codespace in in their repo and  run `pnpm -filter @acme/nextjs add @types/leaflet -D`

Using `pnpm dlx husky-init` fails with the same error so it may be a pnpm thing.

**Comment by juliusmarminge** [comment]
I can confirm `pnpm -F nextjs add -D @types/leaflet`. `pnpm -F nextjs add @types/leaflet` works, and `pnpm -F nextjs add -D @types/leaflet` works if you disable running sherif on postinstall

**Comment by QuiiBz** [comment]
Opened an issue on pnpm's repo to resolve it, since this error occurs with any package and not just sherif: https://github.com/pnpm/pnpm/issues/7424

**Comment by QuiiBz** [comment]
The [fix](https://github.com/pnpm/pnpm/pull/7540) has been released in pnpm [v9.0.0-alpha.1](https://github.com/pnpm/pnpm/releases/tag/v9.0.0-alpha.1) and I've been able to confirm that it works:

<img width="660" alt="Screenshot 2024-01-28 at 09 45 53" src="https://github.com/t3-oss/create-t3-turbo/assets/43268759/e4df8900-86fa-4633-998d-7f08dd3c0c3c">

I'll wait for a stable release of pnpm before closing this issue.

**Comment by QuiiBz** [comment]
Closing this issue as pnpm 9.0.0 has been released a few weeks ago: https://github.com/pnpm/pnpm/releases/tag/v9.0.0
